File: clientSide/src/app/Models/event.ts

export class Event{

    idEvent? : any;
    name : string;
    description : string;
    startDate : Date;
    endDate : Date;
    nbParticipant : number;
    image? : any;
    price: number;
    category: string;
    videoUrl : string;
    idCampPlace?: number;

    constructor(name: string, des: string, startDate: Date, endDate: Date, nbP: number,price: number,cate: string,videoUrl : string ,id?: number,idCampPlace?:number,img?: any) {
        this.idEvent = id;
        this.name = name;
        this.description = des;
        this.startDate = startDate;
        this.endDate = endDate;
        this.nbParticipant = nbP;
        this.price= price;
        this.category=cate;
        this.videoUrl = videoUrl;
      }


}

File: ApiSide/microEvent/src/main/java/org/event/Entity/Event.java
package org.event.Entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idEvent;
    private String name;

    @Column(length = 1000)
    private String description;

    private Date startDate;
    private Date endDate;
    private int nbParticipant;
    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] image;
    private double price;
    private String videoUrl;


    @Enumerated(EnumType.STRING)
    private EventCategory category;



}


